pandas/tests/io/formats/test_style.py

```diff
@@ -87,28 +87,131 @@ def test_update_ctx_flatten_multi_and_trailing_semi(self):
         }
         assert self.styler.ctx == expected
 
-    def test_copy(self):
+    @pytest.mark.parametrize("do_changes", [True, False])
+    @pytest.mark.parametrize("do_render", [True, False])
+    def test_copy(self, do_changes, do_render):
+        # Updated in GH39708
+        # Change some defaults (to check later if they are copied)
+        if do_changes:
+            style = [{"selector": "th", "props": [("foo", "bar")]}]
+            self.styler.set_table_styles(style)
+            attributes = 'class="foo" data-bar'
+            self.styler.set_table_attributes(attributes)
+            self.styler.uuid_len += 1
+            self.styler.hidden_index = not self.styler.hidden_index
+            self.styler.hide_columns("A")
+            classes = pd.DataFrame(
+                [["favorite-val red", ""], [None, "blue my-val"]],
+                index=self.df.index,
+                columns=self.df.columns,
+            )
+            self.styler.set_td_classes(classes)
+            ttips = pd.DataFrame(
+                data=[["Favorite", ""], [np.nan, "my"]],
+                columns=self.df.columns,
+                index=self.df.index,
+            )
+            self.styler.set_tooltips(ttips)
+            self.styler.cell_ids = not self.styler.cell_ids
+            self.styler.format("{:.2%}")
+
+        if do_render:
+            self.styler.render()
+
         s2 = copy.copy(self.styler)
+
+        # Check for identity
         assert self.styler is not s2
         assert self.styler.ctx is s2.ctx  # shallow
         assert self.styler._todo is s2._todo
-
+        assert self.styler.table_styles is s2.table_styles
+        assert self.styler.hidden_columns is s2.hidden_columns
+        assert self.styler.cell_context is s2.cell_context
+        assert self.styler.tooltips is s2.tooltips
+        if do_changes:  # self.styler.tooltips is not None
+            assert self.styler.tooltips.tt_data is s2.tooltips.tt_data
+
+        # Check for equality (and changes in referenced objects)
         self.styler._update_ctx(self.attrs)
         self.styler.highlight_max()
         assert self.styler.ctx == s2.ctx
         assert self.styler._todo == s2._todo
+        assert self.styler.table_styles == s2.table_styles
+        assert self.styler.table_attributes == s2.table_attributes
+        assert self.styler.cell_ids == s2.cell_ids
+        assert self.styler.uuid_len == s2.uuid_len
```

Comment:
`uuid` is the key variable, `uuid_len` is only used in construction of `uuid` test for that instead.